src/main/webapp/script.js

```diff
@@ -0,0 +1,34 @@
+function ratePromoCode(tableRow, rating) {
+    var videoUrl = document.getElementById('promoCodeTable').rows[tableRow].cells[0].firstChild.getAttribute("href");
+    console.log(videoUrl);
+    fetch('/promo-code?rating=' + rating + '&videoUrl=' + videoUrl, {method: 'POST'});
+    getCodes();
+}
+
+function getCodes() {
+    fetch('/promo-code').then(response => response.json()).then((codes) => {
```

Comment:
optional: why not use await vs. .then().  I think await would be easier to read since you don't have to wrap everything inside the brackets.

Reply:
Functionally its the same. This is just the format i'm more comfortable with.

Reply:
should probably change to await for best practice.